pkg/rc/farm.go

```diff
@@ -267,8 +281,21 @@ START_LOOP:
 	}
 }
 
-func (rcf *Farm) failsafe(rcFields []rcstore.RCLockResult) {
-	// FAILSAFES. If no RCs are scheduled, or there are zero replicas of anything scheduled, panic
+// This failsafe is only run at startup of the farm for performance reasons. It checks two conditions:
+// 1) there is at least one RC
+// 2) the sum of the replicas_desired fields for all RCs is greater than zero.
+//
+// If either of those conditions are true, the function panics to prevent the
+// RC farm from uninstalling pods. There is an additional failsafe mechanism in
+// that an RC cannot be unmarshaled from JSON if there is no replicas_desired
+// field which would be the most likely way that all RC counts would be zeroed
+// out.
+func (rcf *Farm) initialFailsafe() {
+	rcFields, err := rcf.rcStore.List()
+	if err != nil {
+		panic(fmt.Sprintf("unable to do initial list of RCs to evaluate failsafe check: %s", err))
+	}
+
 	if len(rcFields) == 0 {
```

Comment:
I wish this could call `failsafe(rcFields)` to avoid duplicate code, but the types are different! Doh.